### PR TITLE
+Refactor internal_tides interface

### PR DESCRIPTION
**Pull request by Hallberg-NOAA**
Refactors the internal tide code in MOM_internal_tides and MOM_diabatic_driver to consolidate it in the MOM_internal_tides module and allow the control structure for that module to be made opaque.  This includes moving the internal wave speed diagnostics and the call to wave_speeds or other code setting the internal wave speeds into propagate_int_tide.  The get_param calls for INTERNAL_WAVE_CG1_THRESH and UNIFORM_TEST_CG were moved from the diabatic module to the MOM_internal_tides module. The wave_speed_CS and uniform_test_cg were removed from diabatic_CS and added to int_tide_CS.  The Nb argument to propagate_int_tide has been made intent inout, as it is now usually set via the call to wave_speeds in that routine, but for certain tests it could use the value passed in from diabatic_driver.

  All answers are bitwise identical, but there are changes to public interfaces and types, and the order of some entries in the MOM_parameter_doc files and the available_diags files is changed for some cases.

### CODE REVIEW

**Comment by raphaeldussin** [comment]
I compared this PR to my refactored branch here https://github.com/raphaeldussin/MOM6/tree/wave_speeds_struct_pr_refactored
and found only minor differences hence I am convinced this PR is sound.

**Review by raphaeldussin — APPROVED** [review]

**Review by marshallward — APPROVED** [review]
Approved on behalf of @raphaeldussin

File: src/parameterizations/lateral/MOM_internal_tides.F90 [hunk 1]
```diff
@@ -2322,8 +2348,7 @@ subroutine internal_tides_init(Time, G, GV, US, param_file, diag, CS)
   use_temperature = .true.
   call read_param(param_file, "ENABLE_THERMODYNAMICS", use_temperature)
   if (.not.use_temperature) call MOM_error(FATAL, &
-    "register_int_tide_restarts: internal_tides only works with "//&
```
Comment (marshallward) [inline src/parameterizations/lateral/MOM_internal_tides.F90:2325]: Not part of this PR, but does `ENABLE_THERMODYNAMICS` need a `do_not_log`?

Comment (marshallward) [reply src/parameterizations/lateral/MOM_internal_tides.F90:2325]: There are other instances of this in the code, and may not even be a problem.  If it is, then we can address this later.

**Comment by marshallward** [comment]
Gaea regression: https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/19654 :heavy_check_mark:
